activiti-core-common/activiti-spring-project/src/main/java/org/activiti/core/common/spring/project/ApplicationUpgradeContextService.java

```diff
@@ -45,11 +49,19 @@ public ProjectManifest loadProjectManifest() throws IOException {
         Optional<Resource> resourceOptional = retrieveResource();
 
         return read(resourceOptional
-                            .orElseThrow(() -> new FileNotFoundException("'" + projectManifestFilePath + "' manifest not found."))
-                            .getInputStream());
+                        .orElseThrow(() -> new FileNotFoundException("'" + projectManifestFilePath + "' manifest not found."))
+                        .getInputStream());
     }
 
-    public boolean hasProjectManifest(){
+    public boolean hasProjectManifest() {
         return retrieveResource().isPresent();
     }
+
+    public boolean hasEnforcedAppVersion() {
+        return this.enforcedAppVersion != 0;
```

Comment:
would > 0 be better in case of negative app version passed?

Reply:
This enforced version is set by the deployment service, and it is a counter AFAIK
But we can add that check just in case the env variable is set via other mechanisms we don't control/expect